src/trust.rs

```diff
@@ -144,6 +148,29 @@ pub fn create_device_certificate(
     Ok(())
 }
 
+fn verify_public_key(ca_cert: &str, local_cert: &[u8]) -> Result<()> {
+    let ca_x509 = x509_parser::pem::parse_x509_pem(&ca_cert.as_bytes())?.1;
+    let ca_x509_der = x509_parser::parse_x509_certificate(&ca_x509.contents)?.1;
+
+    let local_certificate = x509_parser::parse_x509_certificate(&local_cert)?.1;
+
+    let ca_public_key = ca_x509_der
+        .tbs_certificate
+        .subject_pki
+        .subject_public_key
+        .data;
+    let local_public_key = local_certificate
+        .tbs_certificate
+        .subject_pki
+        .subject_public_key
+        .data;
+
+    ca_public_key
+        .eq(local_public_key)
+        .then(|| ())
+        .ok_or(anyhow!("Wrong CA key"))
```

Comment:
This error should be more helpful for the user. "Invalid" is more appropriate than "Wrong" in this context.

Reply:
It is a valid key but just not the correct one, so let's go with `Invalid CA key: trust anchor and private key mismatch`